File: Java/0012.java
class Solution {
    public String intToRoman(int num) {
        // Define the symbols and their corresponding values
        int[] values = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] symbols = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        
        StringBuilder result = new StringBuilder();
        
        // Iterate over the values and symbols
        for (int i = 0; i < values.length; i++) {
            // Determine how many times the current value fits into num
            while (num >= values[i]) {
                // Append the corresponding symbol
                result.append(symbols[i]);
                // Reduce num by the value
                num -= values[i];
            }
        }
        
        return result.toString();
    }

}

File: Kotlin/0012.kt
class Solution {
    fun intToRoman(num: Int): String {
        // Define the values and symbols
        val values = intArrayOf(1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1)
        val symbols = arrayOf("M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I")
        
        val result = StringBuilder()
        
        var n = num
        
        // Iterate over the values and symbols
        for (i in values.indices) {
            // Determine how many times the current value fits into num
            while (n >= values[i]) {
                // Append the corresponding symbol
                result.append(symbols[i])
                // Reduce num by the value
                n -= values[i]
            }
        }
        
        return result.toString()
    }
}
